cmd/root.go

```diff
@@ -358,44 +374,65 @@ func updateIssue(ghIssue github.Issue, jIssue jira.Issue, client jira.Client) er
 
 	key = fmt.Sprintf("customfield_%s", ghLabelsFieldID)
 	field, err = jIssue.Fields.Unknowns.String(key)
-	if err != nil || strings.Join(labels, ",") != field {
+	if err != nil && strings.Join(labels, ",") != field {
 		anyDifferent = true
 		fields.Unknowns[key] = strings.Join(labels, ",")
 	}
 
-	if !anyDifferent {
-		log.Debugf("JIRA issue %s is already up to date!", jIssue.Key)
-		return nil
-	}
+	if anyDifferent {
+		key = fmt.Sprintf("customfield_%s", isLastUpdateFieldID)
+		fields.Unknowns[key] = time.Now().Format(dateFormat)
 
-	key = fmt.Sprintf("customfield_%s", isLastUpdateFieldID)
-	fields.Unknowns[key] = time.Now().Format("2006-01-02T15:04:05-0700")
+		fields.Type = jIssue.Fields.Type
+		fields.Summary = jIssue.Fields.Summary
 
-	fields.Type = jIssue.Fields.Type
-	fields.Summary = jIssue.Fields.Summary
+		issue := &jira.Issue{
+			Fields: &fields,
+			Key:    jIssue.Key,
+			ID:     jIssue.ID,
+		}
 
-	issue := &jira.Issue{
-		Fields: &fields,
-		Key:    jIssue.Key,
-		ID:     jIssue.ID,
-	}
+		issue, res, err := jClient.Issue.Update(issue)
+
+		if err != nil {
+			log.Errorf("Error updating JIRA issue %s: %v", jIssue.Key, err)
+			defer res.Body.Close()
+			body, _ := ioutil.ReadAll(res.Body)
+			log.Debugf("Error body: %s", body)
+			return err
+		}
 
-	issue, res, err := client.Issue.Update(issue)
+		log.Debugf("Successfully updated JIRA issue %s!", jIssue.Key)
+	} else {
+		log.Debugf("JIRA issue %s is already up to date!", jIssue.Key)
+	}
 
+	issue, _, err := jClient.Issue.Get(jIssue.ID, nil)
 	if err != nil {
-		log.Errorf("Error updating JIRA issue %s: %s", jIssue.Key, err)
-		defer res.Body.Close()
-		body, _ := ioutil.ReadAll(res.Body)
-		log.Debugf("Error body: %s", body)
-		return err
+		log.Errorf("Error retrieving JIRA issue %s to get comments.", jIssue.Key)
 	}
 
-	log.Debugf("Successfully updated JIRA issue %s!", jIssue.Key)
+	var comments []jira.Comment
+	if issue.Fields.Comments == nil {
+		log.Debugf("JIRA issue %s has no comments.", jIssue.Key)
+		comments = nil
```

Comment:
Oops. Yep, that's a holdover from C programming, and the assumption that `var comments []jira.Comment` will leave comments uninitialized and unusable.